src/backend/cuda/subconfigure.m4

```diff
@@ -61,51 +61,55 @@ if test "$with_cuda" != "no" ; then
     PAC_CHECK_HEADER_LIB([cuda_runtime_api.h],[cudart],[cudaStreamSynchronize],[have_cuda=yes],[have_cuda=no])
     if test "${have_cuda}" = "yes" ; then
         AC_MSG_CHECKING([whether nvcc works])
-        cat>conftest.cu<<EOF
-        __global__ void foo(int x) {}
-EOF
         if test -n "$ac_save_CC" ; then
-            NVCC_FLAGS="-ccbin $ac_save_CC"
+            NVCC_FLAGS="$NVCC_FLAGS -ccbin $ac_save_CC"
             # - pgcc/nvc doesn't work, use pgc++/nvc++ instead
             # - Extra optins such as `gcc -std=gnu99` doesn't work, strip the option
             NVCC_FLAGS=$(echo $NVCC_FLAGS | sed -e 's/nvc/nvc++/g' -e 's/pgcc/pgc++/g' -e's/ -std=.*//g')
-        else
-            NVCC_FLAGS=""
         fi
         # try nvcc from PATH if 'with-cuda' does not contain a valid path
         if test -d ${with_cuda} ; then
             nvcc_bin=${with_cuda}/bin/nvcc
         else
             nvcc_bin=nvcc
         fi
-        ${nvcc_bin} $NVCC_FLAGS -c conftest.cu 2> /dev/null
-        if test "$?" = "0" ; then
+
+        # save language settings, customize ac_ext and ac_compile to support CUDA
+        AC_LANG_PUSH([C])
+        ac_ext=cu
+        ac_compile='$nvcc_bin $NVCC_FLAGS -c conftest.$ac_ext >&5'
+        AC_COMPILE_IFELSE([AC_LANG_PROGRAM([__global__ void foo(int x) {}],[])],
+        [
             AC_DEFINE([HAVE_CUDA],[1],[Define is CUDA is available])
-            AS_IF([test -n "${with_cuda}"],[NVCC=${with_cuda}/bin/nvcc],[NVCC=nvcc])
+            AS_IF([test -d "${with_cuda}"],[NVCC=${with_cuda}/bin/nvcc],[NVCC=nvcc])
             AC_SUBST(NVCC)
             AC_SUBST(NVCC_FLAGS)
```

Comment:
Both AC_SUBST probably should move out since they are not part of the IFELSE logic.

Reply:
I'll save this one for another PR. I notice that yaksa needs an update to silence all the autoconf deprecated warnings. Will handle together with that.